src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -1,41 +1,144 @@
+import com.revrobotics.EncoderType;
 import com.revrobotics.CANSparkMaxLowLevel.MotorType;
 
 import edu.wpi.first.wpilibj.SpeedControllerGroup;
 import edu.wpi.first.wpilibj.drive.DifferentialDrive;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-
+import frc.robot.Constants;
 import frc.robot.Constants.Ports;
 
+
 /**
  * Drivetrain
  */
 public class Drivetrain extends SubsystemBase {
+    // in inches
+    public static final double WHEEL_CIRCUMFERENCE = 6 * Math.PI;
+    public static final int NEO_ENCODER_CPR = 4096;
+    public static final int NEO_ENCODER_PPR = 4096 / 4;
+    public static final int SRX_ENCODER_PPR = 4096 / 4;
 
     private final CANSparkMax m_frontLeftMotor = new CANSparkMax(Ports.FRONT_LEFT_MOTOR_PORT, MotorType.kBrushless);
     private final CANSparkMax m_frontRightMotor = new CANSparkMax(Ports.FRONT_RIGHT_MOTOR_PORT, MotorType.kBrushless);
     private final CANSparkMax m_rearLeftMotor = new CANSparkMax(Ports.REAR_LEFT_MOTOR_PORT, MotorType.kBrushless);
     private final CANSparkMax m_rearRightMotor = new CANSparkMax(Ports.REAR_RIGHT_MOTOR_PORT, MotorType.kBrushless);
 
+    private final WPI_TalonSRX m_leftSRXEncoderMotor = new WPI_TalonSRX(Ports.LEFT_ENCODER_MOTOR);
+    private final WPI_TalonSRX m_rightSRXEncoderMotor = new WPI_TalonSRX(Ports.RIGHT_ENCODER_MOTOR);
+
     private final SpeedControllerGroup m_leftSide = new SpeedControllerGroup(m_frontLeftMotor, m_rearLeftMotor);
     private final SpeedControllerGroup m_rightSide = new SpeedControllerGroup(m_frontRightMotor, m_rearRightMotor);
     
     private final DifferentialDrive m_drivetrain = new DifferentialDrive(m_leftSide, m_rightSide);
 
+    private final CANEncoder m_leftNeoEncoder = m_frontLeftMotor.getEncoder(EncoderType.kQuadrature, NEO_ENCODER_CPR);
+    private final CANEncoder m_rightNeoEncoder = m_frontRightMotor.getEncoder(EncoderType.kQuadrature, NEO_ENCODER_CPR);
     private final double m_motor_deadband = 0.05;
 
-    public Drivetrain() {
+    public Drivetrain(EncoderBrand encoderBrand) {
         m_drivetrain.setDeadband(m_motor_deadband);
         m_drivetrain.setSafetyEnabled(true);
+        
+        // set up encoders
+        EncoderBrand.setCurrent(encoderBrand);
+        m_leftNeoEncoder.setPositionConversionFactor(WHEEL_CIRCUMFERENCE / m_leftNeoEncoder.getCountsPerRevolution());
+        m_rightNeoEncoder.setPositionConversionFactor(WHEEL_CIRCUMFERENCE / m_rightNeoEncoder.getCountsPerRevolution());
+        m_leftSRXEncoderMotor.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative);
+        m_rightSRXEncoderMotor.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative);
+        resetEncoders();
+        
+        // for testing
+        // setMaxOutput(0.2);
+    }
+
+    public enum EncoderBrand {
+        NEO, SRX;
+
+        private static EncoderBrand current = NEO;
```

Comment:
You guys did this last year too. This is not how you should do it. Remove all the static methods here.